wandb/apis/public.py

```diff
@@ -287,15 +287,21 @@ class Api(object):
     )
 
     def __init__(self, overrides={}, timeout: Optional[int] = None):
-        self.settings = InternalApi().settings()
-        if self.api_key is None:
-            wandb.login()
-        self.settings.update(overrides)
+
         if "username" in overrides and "entity" not in overrides:
             wandb.termwarn(
                 'Passing "username" to Api is deprecated. please use "entity" instead.'
             )
-            self.settings["entity"] = overrides["username"]
+            overrides["entity"] = overrides["username"]
+
+        _api = InternalApi()
+        for key, value in overrides.items():
+            _api.set_setting(key, value)
+        self.settings = _api.settings()
+
+        if self.api_key is None:
+            wandb.login(host=self.settings["base_url"])
```

Comment:
This change overrides the global settings!!! The proper solution would be to use the `wandb.Settings`,  this will a big refactoring work.